nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/test/java/org/apache/nifi/elasticsearch/integration/ElasticSearchClientService_IT.java

```diff
@@ -205,19 +206,29 @@ void testVerifyFailedApiKeyAuth() {
 
     @Test
     void testBasicSearch() throws Exception {
+        assertBasicSearch(null);
+    }
+
+    @Test
+    void testBasicSearchRequestParameters() throws Exception {
+        assertBasicSearch(createParameters("preference", "_local"));
+    }
+
+    private void assertBasicSearch(final Map<String, String> requestParameters) throws JsonProcessingException {
         final Map<String, Object> temp = new MapBuilder()
-            .of("size", 10, "query", new MapBuilder().of("match_all", new HashMap<>()).build(),
-                    "aggs", new MapBuilder()
-                            .of("term_counts", new MapBuilder()
-                                    .of("terms", new MapBuilder()
-                                            .of("field", "msg", "size", 5)
-                                            .build())
-                                    .build())
-                            .build())
+                .of("size", 10, "query", new MapBuilder().of("match_all", new HashMap<>()).build(),
```

Comment:
Recommend using `LinkedHashMap`:
```suggestion
                .of("size", 10, "query", new MapBuilder().of("match_all", new LinkedHashMap<>()).build(),
```

Reply:
I don't think this is required, the use of `HashMap` here (and throughout all uses of the `MapBuilder` class in the tests) has been to create an empty Map (i.e. JSON Object) for the `match_all` part of the Elasticsearch Query DSL request body

I have, however, changed the `MapBuilder` to use a `LinkedHashMap` internally instead of a `HashMap`, which might be relevant for keeping things in order (although that shouldn't matter for the submission to Elasticsearch)